pkg/apis/gcp/validation/cloudprofile.go

```diff
@@ -19,37 +19,50 @@ import (
 
 	apisgcp "github.com/gardener/gardener-extension-provider-gcp/pkg/apis/gcp"
 
+	"github.com/gardener/gardener/pkg/apis/core"
 	"k8s.io/apimachinery/pkg/util/validation/field"
 )
 
 // ValidateCloudProfileConfig validates a CloudProfileConfig object.
-func ValidateCloudProfileConfig(cloudProfile *apisgcp.CloudProfileConfig) field.ErrorList {
+func ValidateCloudProfileConfig(cpConfig *apisgcp.CloudProfileConfig, machineImages []core.MachineImage, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
+	machineImagesPath := fldPath.Child("machineImages")
 
-	machineImagesPath := field.NewPath("machineImages")
-	if len(cloudProfile.MachineImages) == 0 {
-		allErrs = append(allErrs, field.Required(machineImagesPath, "must provide at least one machine image"))
+	for _, image := range machineImages {
+		var processed bool
+		for i, imageConfig := range cpConfig.MachineImages {
+			if image.Name == imageConfig.Name {
+				allErrs = append(allErrs, ValidateVersions(imageConfig.Versions, image.Versions, machineImagesPath.Index(i).Child("versions"))...)
+				processed = true
+				break
+			}
+		}
+		if !processed && len(image.Versions) > 0 {
+			allErrs = append(allErrs, field.Required(machineImagesPath, fmt.Sprintf("must provide an image mapping for image %q", image.Name)))
+		}
 	}
-	for i, machineImage := range cloudProfile.MachineImages {
-		idxPath := machineImagesPath.Index(i)
 
-		if len(machineImage.Name) == 0 {
-			allErrs = append(allErrs, field.Required(idxPath.Child("name"), "must provide a name"))
-		}
+	return allErrs
+}
 
-		if len(machineImage.Versions) == 0 {
-			allErrs = append(allErrs, field.Required(idxPath.Child("versions"), fmt.Sprintf("must provide at least one version for machine image %q", machineImage.Name)))
-		}
-		for j, version := range machineImage.Versions {
-			jdxPath := idxPath.Child("versions").Index(j)
+func ValidateVersions(versionsConfig []apisgcp.MachineImageVersion, versions []core.ExpirableVersion, fldPath *field.Path) field.ErrorList {
```

Comment:
Can you add a doc string for this exported function?

Reply:
I made it private since it's not required by an external package.